zerver/worker/queue_processors.py

```diff
@@ -118,32 +118,29 @@ class WorkerDeclarationException(Exception):
 ConcreteQueueWorker = TypeVar('ConcreteQueueWorker', bound='QueueProcessingWorker')
 
 def assign_queue(
-        queue_name: str, enabled: bool=True, queue_type: str="consumer",
+        queue_name: str, enabled: bool=True, is_test_queue: bool=False,
 ) -> Callable[[Type[ConcreteQueueWorker]], Type[ConcreteQueueWorker]]:
     def decorate(clazz: Type[ConcreteQueueWorker]) -> Type[ConcreteQueueWorker]:
         clazz.queue_name = queue_name
         if enabled:
-            register_worker(queue_name, clazz, queue_type)
+            register_worker(queue_name, clazz, is_test_queue)
         return clazz
     return decorate
 
 worker_classes: Dict[str, Type["QueueProcessingWorker"]] = {}
-queues: Dict[str, Dict[str, Type["QueueProcessingWorker"]]] = {}
-def register_worker(queue_name: str, clazz: Type['QueueProcessingWorker'], queue_type: str) -> None:
-    if queue_type not in queues:
-        queues[queue_type] = {}
-    queues[queue_type][queue_name] = clazz
+test_queues: Set[str] = set()
+def register_worker(queue_name: str, clazz: Type['QueueProcessingWorker'], is_test_queue: bool=False) -> None:
     worker_classes[queue_name] = clazz
+    if is_test_queue:
+        test_queues.add(queue_name)
 
 def get_worker(queue_name: str) -> 'QueueProcessingWorker':
     return worker_classes[queue_name]()
 
-def get_active_worker_queues(queue_type: Optional[str]=None, include_test_queues: bool=False) -> List[str]:
-    """Returns all the non-test worker queues."""
-    if queue_type is None:
-        return [queue_name for queue_name in worker_classes.keys()
-                if include_test_queues or queue_name not in queues["test"]]
-    return list(queues[queue_type].keys())
+def get_active_worker_queues(only_test_queues: bool=False) -> List[str]:
```

Comment:
This argument name seems a bit confusing to me, because the `only` part suggests that with the flag `Falses` we'd return real+test queues (because *not only* test queues), and with `True` just the test queues.

Reply:
Hm -- I can see that parsing of it, though it clearly wasn't my intended one.  Would it help to split the function into `get_active_worker_queues() -> List[str]` and `get_test_queues() -> List[str]` ?